larray/core/array.py

```diff
@@ -3978,7 +3978,7 @@ def wrapper(self, *args, **kwargs):
         return decorated
 
     @_decorate_agg_method(np.all, commutative=True, long_name="AND reduction")
-    def all(self, *args, **kwargs):
+    def all(self, *args, **kwargs) -> 'Array':
```

Comment:
@gdementen I have a doubt about the 'declared' returned type of aggregation functions.
Since the goal is to allow auto-completion in PyCharm and especially on Array objects, I declared the returned type to be only `Array` while the real returned type is Union[Scalar, Array]. I bet that aggregation functions will return an Array in more than 90% of the time in our in-house models. But maybe this a bad guess.

I'm totally OK to switch to Union[Scalar, Array] if you think it is a better choice.

Reply:
If completion works if you use Union[Scalar, Array], then I see no reason (at this point -- there might be one) not to use that. If it does not, that's super annoying and I guess "-> Array:  # or scalar" would be a sort of compromise.

Reply:
Well, surprisingly, completion in PyCharm works like this:
```python
agg_func(...) -> Union[Array, str]:
    # body of the function

arr.agg_func(...).<union_of_methods_of_Array_and_str>
```
So, I changed all returned type of aggregation functions/methods to Union[Array, Scalar]